x/crosschain/keeper/evm_hooks.go

```diff
@@ -267,38 +250,25 @@ func (k Keeper) ProcessZetaSentEvent(
 		event.Raw.Index,
 	)
 
-	// create a new cctx with status as pending Inbound,
-	// this is created directly from the event without waiting for any observer votes
-	cctx, err := types.NewCCTX(ctx, *msg, tss.TssPubkey)
+	cctx, err := k.ValidateInboundObservers(ctx, msg, true)
 	if err != nil {
-		return fmt.Errorf("ProcessZetaSentEvent: failed to initialize cctx: %s", err.Error())
+		return err
 	}
-	cctx.SetPendingOutbound("ZetaSent event setting to pending outbound directly")
 
-	if err := k.PayGasAndUpdateCctx(
-		ctx,
-		receiverChain.ChainId,
-		&cctx,
-		amount,
-		true,
-	); err != nil {
-		return fmt.Errorf("ProcessWithdrawalEvent: pay gas failed: %s", err.Error())
+	if cctx.CctxStatus.Status == types.CctxStatus_Aborted {
+		return errors.New("cctx aborted")
 	}
 
-	EmitZetaWithdrawCreated(ctx, cctx)
-	return k.ProcessCCTX(ctx, cctx, receiverChain)
+	EmitZetaWithdrawCreated(ctx, *cctx)
+	return k.ProcessCCTX(ctx, *cctx)
```

Comment:
This function is used to call
```
k.SetCctxAndNonceToCctxAndInboundHashToCctx(ctx, cctx)
```
same as `SaveInbound`
I think this logic could be put in `ValidateInbound` in common and remove `ProcessCCTX`

Reply:
I meant to keep `SaveInbound` but moving `SetCctxAndNonceToCctxAndInboundHashToCctx` in common